### PR TITLE
Assessments structure API

### PR DESCRIPTION
**Pull request by caleybrock**
## Summary
The new assessments tab (behind internal only experiment) requires information of the structures of the assessments. The question and potentially answer text will get displayed in a table next to the actual student responses, which are retrieved in a separate API call.

The object should return something like this:
```
{
  2345: {   #a level id associated with an assessment
    id: 2345,
    name: "Assessment for Chapter 1",
    questions: {123: {type: "Multi", question_text: "A question", answers: [{text: "answer1", correct: true}] }}
  }
  ..other assessment ids
}
```

assessments_structure

## Where are the tests?
This definitely needs controller tests. I want to do this before the end of the week, but I wanted to have something to unblock Nkiru with on the front end. I added a todo in code for this so that I'm on the hook. 

## Future thoughts
This does end up being a fairly similar structure (although different data) to `section_assessments` which is used to get student responses on the current assessments tab. 
https://github.com/code-dot-org/code-dot-org/blob/staging/dashboard/app/controllers/api_controller.rb#L409
I plan on not touching `section_assessments`, and only adding other APIs, so that the current assessments tab remains unchanged while the new one is in development. It's possible I create another API for student responses that returns a hash of students to levels to responses to make look ups easier. `section_assessments` has an array of responses which might be harder to work with.

Up next: calling this API and adding it to redux (looks like this)
screen shot 2018-05-17 at 4 38 55 pm

### CODE REVIEW

File: dashboard/app/controllers/api_controller.rb [hunk 1]
```diff
@@ -504,6 +504,64 @@ def section_assessments
...
+      questions = {}
+
+      # For each level in the multi group (ignore pages structure information)
+      level_group.levels.each do |level|
+        # A single level corresponds to a single question
+        question = {}
+
+        case level
+        when FreeResponse
+          puts level.properties
```
Comment (breville) [inline dashboard/app/controllers/api_controller.rb]: nit: `puts`

**Comment by breville** [comment]
Getting a stub API up and running quickly to unblock UI development is a great strategy.

**Comment by caleybrock** [comment]
Here's my diagram of all the models.
script models

File: dashboard/app/controllers/api_controller.rb [hunk 2]
```diff
@@ -504,6 +504,63 @@ def section_assessments
...
+
+      # For each level in the multi group (ignore pages structure information)
+      level_group.levels.each do |level|
+        # A single level corresponds to a single question
+        question = {}
+
+        case level
+        when FreeResponse
+          question = {
+            type: level.type,
```
Comment (balderdash) [inline dashboard/app/controllers/api_controller.rb:540]: I prefer to make use of polymorphism than switch on class type. The code to create these hashes could live entirely in the `Multi` and `FreeResponse` classes, and this loop would just be:
```
level_group.levels.each do |level|
  questions[level.id] = level.question_summary
end
```

Comment (caleybrock) [reply dashboard/app/controllers/api_controller.rb:540]: I'll follow up on this.

**Pull request merged**